### PR TITLE
Kuznietsov Valerii homework 1-3

### PR DESCRIPTION
**Pull request by jasketikus**

### CODE REVIEW

File: Valerii/HW1/Practical_Task1_Valerii.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name = input('"WHAT IS YOUR NAME?"\n')
```
Comment (kolyasalubov) [inline Valerii/HW1/Practical_Task1_Valerii.py:1]: good job

File: Valerii/HW1/Practical_Task2_Valerii.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+a = int(input("A = "))
```
Comment (kolyasalubov) [inline Valerii/HW1/Practical_Task2_Valerii.py:1]: good job

File: Valerii/HW3/1_Python_philosophy.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import codecs
+import this
+
```
Comment (kolyasalubov) [inline Valerii/HW3/1_Python_philosophy.py:3]: good job

File: Valerii/HW3/3_variables_swap.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+a = "Hello"
```
Comment (kolyasalubov) [inline Valerii/HW3/3_variables_swap.py:1]: good job

File: Valerii/HW3/2_four_digit.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+number = 1443
```
Comment (kolyasalubov) [inline Valerii/HW3/2_four_digit.py]: good job
but please look at this implementation:
number = int(input("Enter your number: "))

one_div, remainder = divmod(number, 1000)
two_div, remainder = divmod(remainder, 100)
three_div, four_div = divmod(remainder, 10)

#### Item 1: find of product of the digits of this number:
prod_of_dig = one_div * two_div * three_div * four_div
print("Product of the digits of this number:", prod_of_dig)

##### Item 2: write the number in reverse order:
str_number = str(number)[::-1]
print("Reverse of our number:", str_number)

##### Item 3: in ascending order, you need to sort the number included in the given number:
sort_number = ''.join(sorted(str(number)))
print("Ascending order of digits in the number:", sort_number)

Comment (jasketikus) [reply Valerii/HW3/2_four_digit.py]: Ty for review, resolved

File: Valerii/HW3/2_four_digit.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+# for i in number_string:
+#     product_of_digits *= int(i)
+# print(product_of_digits)
+# number_reversed = number_string[::-1]
+# print(number_reversed)
+# number_sorted = sorted(str(number))
+# print(number_sorted)
+
+#Second version:
+
```
Comment (kolyasalubov) [inline Valerii/HW3/2_four_digit.py:13]: good job

File: Valerii/HW4/Temp_converter.py [hunk 7]
```diff
@@ -0,0 +1,7 @@
+celsius = int(input('Enter temperature in Celsius: '))
```
Comment (kolyasalubov) [inline Valerii/HW4/Temp_converter.py:1]: good job

File: Valerii/HW5/task1.py [hunk 8]
```diff
@@ -0,0 +1,4 @@
+l = [int(input('Enter integer number: ')) for number in range(int(input('Enter numbers quantity: ')))]
```
Comment (kolyasalubov) [inline Valerii/HW5/task1.py:1]: good job

File: Valerii/HW5/task2.py [hunk 9]
```diff
@@ -0,0 +1,15 @@
+fib_len = int(input('Enter Fibonacci list lenght: '))
```
Comment (kolyasalubov) [inline Valerii/HW5/task2.py:1]: good job

File: Valerii/HW5/task3.py [hunk 10]
```diff
@@ -0,0 +1,11 @@
+number = int(input('Enter your number to know factorial: '))
```
Comment (kolyasalubov) [inline Valerii/HW5/task3.py:1]: good job

File: Valerii/HW6/task1.py [hunk 11]
```diff
@@ -0,0 +1,14 @@
+even_numbers = []
```
Comment (kolyasalubov) [inline Valerii/HW6/task1.py:1]: good job

File: Valerii/HW6/task2.py [hunk 12]
```diff
@@ -0,0 +1,5 @@
+login = input('Hello, enter login: ')
```
Comment (kolyasalubov) [inline Valerii/HW6/task2.py:1]: good job

please look at this implementation:

user_name = input('Hello, please input your Log in:')

while user_name != 'First':
    user_name = input('Error: wrong username, please try one more time. Username:')
else:
    print('Greeting. Access granted!!!', user_name)

Comment (kolyasalubov) [inline Valerii/HW6/task2.py:1]: good job

File: Valerii/HW7/task1.py [hunk 13]
```diff
@@ -0,0 +1,9 @@
+def largest_number(a: int, b: int) ->int:
```
Comment (kolyasalubov) [inline Valerii/HW7/task1.py:1]: good job

File: Valerii/HW7/task2.py [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+
+def area_circle(r):
+    """ This function calculates the area of circle with radius 'r'."""
+    return pi*r**2
+
+def main():
+    """ The main function, that calculates the area of rectangle, triangle or circle
+    depending on the user's choise.    
+    """
+    choise = input('Hello, what area do you want to calculate?\nEnter rectange/triangle/circle: ')
```
Comment (kolyasalubov) [inline Valerii/HW7/task2.py:24]: choise = input('Hello, what area do you want to calculate?\nEnter rectange/triangle/circle: ').lower()

File: Valerii/HW7/task3.py [hunk 15]
```diff
@@ -0,0 +1,9 @@
+def char_numbers(s: str) -> dict:
```
Comment (kolyasalubov) [inline Valerii/HW7/task3.py:1]: good job

File: Valerii/HW8/areas.py [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+
+def area_circle(r):
+    """ This function calculates the area of circle with radius 'r'."""
+    return pi*r**2
+
+def calculate_throuht_input():
+    """ The main function, that calculates the area of rectangle, triangle or circle
+    depending on the user's choise.    
+    """
+    choise = input('Hello, what area do you want to calculate?\nEnter rectange/triangle/circle: ')
```
Comment (kolyasalubov) [inline Valerii/HW8/areas.py:24]: choise = input('Hello, what area do you want to calculate?\nEnter rectange/triangle/circle: ').lower()

File: Valerii/HW8/areas.py [hunk 17]
```diff
@@ -0,0 +1,38 @@
+from math import pi
```
Comment (kolyasalubov) [inline Valerii/HW8/areas.py:1]: good job

File: Valerii/HW8/practical_task_1/main.py [hunk 18]
```diff
@@ -0,0 +1,4 @@
+from models import *
```
Comment (kolyasalubov) [inline Valerii/HW8/practical_task_1/main.py:1]: good job

File: Valerii/HW8/practical_tast_2.py [hunk 19]
```diff
@@ -0,0 +1,31 @@
+def password_validator(password):
```
Comment (kolyasalubov) [inline Valerii/HW8/practical_tast_2.py:1]: good job

File: Valerii/HW8/practical_tast_2.py [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+    has_digits = any(char.isdigit() for char in password)
```
Comment (kolyasalubov) [inline Valerii/HW8/practical_tast_2.py:2]: look at this implementation

import re

def password_validator(password):
    # Check length
    if len(password) < 6 or len(password) > 16:
        print(f'Password length should be between 6 and 16 characters. You entered {len(password)} characters.')
        return False

    # Check if password contains at least one digit, lowercase letter, uppercase letter, and special character
    if not re.search(r'\d', password):
        print('Password should contain at least one digit.')
        return False

    if not re.search(r'[a-z]', password):
        print('Password should contain at least one lowercase letter.')
        return False

    if not re.search(r'[A-Z]', password):
        print('Password should contain at least one uppercase letter.')
        return False

    if not re.search(r'[$#@]', password):
        print('Password should contain at least one of the following special characters: $ # @')
        return False

    return True

if __name__ == '__main__':
    password = input('Enter your password: ')
    while not password_validator(password):
        password = input('Try again: ')
    print('Your password is confirmed.')

**Pull request merged**